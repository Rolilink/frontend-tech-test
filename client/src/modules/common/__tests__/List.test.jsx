import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import List from '../List';

expect.addSnapshotSerializer(serializer);
const renderListItem = item => <div className="list-item" key={`list-item-${item}`}>{item}</div>;

describe('Common.List', () => {
  it('should call itemRenderer to render items', () => {
    const renderListItemMock = jest.fn(renderListItem);
    const items = [1, 2];
    shallow(<List renderListItem={renderListItemMock} items={items} />);

    expect(renderListItemMock.mock.calls.length).toBe(2);
  });

  it('should render items', () => {
    const items = [1, 2];
    const list = shallow(<List renderListItem={renderListItem} items={items} />);

    expect(list).toMatchSnapshot();
  });

  it('should not render items when empty', () => {
    const items = [];
    const list = shallow(<List renderListItem={renderListItem} items={items} />);

    expect(list).toMatchSnapshot();
  });
});
