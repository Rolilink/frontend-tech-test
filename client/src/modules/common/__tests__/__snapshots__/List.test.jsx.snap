// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common.List should not render items when empty 1`] = `
<ListGroup
  bsClass="list-group"
  className=""
/>
`;

exports[`Common.List should render items 1`] = `
<ListGroup
  bsClass="list-group"
  className=""
>
  <div
    className="list-item"
  >
    1
  </div>
  <div
    className="list-item"
  >
    2
  </div>
</ListGroup>
`;
